avl_template_new.py

```diff
@@ -1072,21 +1072,45 @@ def shuffle(self, array):
 	@returns: the absolute value of the difference between the height of the AVL trees joined
 	"""
 	def concat(self, lst):
-		absDiff = abs(self.root.height - lst.root.height)
+		absDiff = abs(self.getRoot().getHeight() - lst.getRoot().getHeight())
+		shorterTree = self
+		LongerTree = lst
+
+		b = LongerTree.getRoot()
+		while b.getHeight() > shorterTree.getRoot().getHeight():
+			b = b.getLeft()
```

Comment:
what if b has only a right child?

Reply:
we are going over the laft side of the tree only, the algo will know how to deal if the original list is is the bigger tree